mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -436,39 +438,48 @@ fn find_pointers_and_order_post_crash(file: &mut File, max_file_size: u64) -> Ri
             read = end + data_size;
         }
 
-        // Found the last write, take whatever we got for read and write
-        // and return the pointers.
+        // Check next block
+        write_updates += 1;
+        order += 1; // We want to start with the next order number.
+        start = end + data_size;
+        end = start + block_size;
+
+        // Found the last write, can stop updating write.
         if inner.order() < order {
+            should_update_write = false;
+            // Update write one last time to point at block
+            // where order is less.
+            write = start % max_file_size;
```

Comment:
So, this particular case is as follows:
1. Fill RingBuffer entirely, except small portion that cannot fit next amount of data
2. Read and delete first entry so that new entry will fit
3. Write entry, where write pointer is now at beginning area of the buffer and 'before' read pointer
4. crash
5. Initialization will read the whole file and find the last write at the end of the buffer, but that the write pointer should wrap around and start after that last entry

Can also draw out any of these cases if they help :)